packages/@aws-cdk/aws-rds/lib/proxy.ts

```diff
@@ -412,12 +412,16 @@ export class DatabaseProxy extends cdk.Resource
     let dbInstanceIdentifiers: string[] | undefined;
     if (bindResult.dbInstances) {
       // support for only single instance
-      dbInstanceIdentifiers = [bindResult.dbInstances[0].instanceIdentifier];
+      const dbInstance = bindResult.dbInstances[0];
+      this.node.addDependency(dbInstance);
+      dbInstanceIdentifiers = [dbInstance.instanceIdentifier];
```

Comment:
This is not needed. Since the `ProxyTargetGroup` depends on the Instance already (through the `dbInstance.instanceIdentifier` expression), and also depends on `Proxy` itself through `this.dbProxyName`, this doesn't change anything as far as the dependency order is concerned.

Revert these changes please.